components/client/src/main/java/com/hotels/styx/client/ConnectionSettings.java

```diff
@@ -71,6 +71,8 @@ public int connectTimeoutMillis() {
     /**
      * Deprecated and due to be removed in a future release.
      *
+     * @deprecated In the next version.
```

Comment:
I have added the deprecation tag in this commit to keep static analysis happy. Otherwise the change is not related to this pull request.

For what I can remember is that `socketTimeout` is relevant for blocking APIs only. A waiting thread is unblocked if the connection doesn't complete within this time. IIRC it is not relevant for non-blocking APIs.

If you want this to be rectified in the comments, please raise as a separate GitHub issue.